uw_oidc/middleware.py

```diff
@@ -25,34 +29,35 @@ def process_view(self, request, view_func, view_args, view_kwargs):
 
         if 'HTTP_AUTHORIZATION' in request.META:
             try:
+                if request.user.is_authenticated:
+                    # not validate ID token on every request
+                    return None
+
+                # We are seeing this user for the first time in this
+                # session, attempt to authenticate the user.
                 token = request.META['HTTP_AUTHORIZATION']
                 username = self.clean_username(
                     UWIdPToken().username_from_token(token))
 
-                if request.user.is_authenticated:
-                    if request.user.get_username() != username:
-                        # An authenticated user is associated with the request,
-                        # but it does not match the user in the token.
-                        raise InvalidTokenError('Username mismatch')
-                else:
-                    # We are seeing this user for the first time in this
-                    # session, attempt to authenticate the user.
-                    user = auth.authenticate(request, remote_user=username)
-                    if user:
-                        # User is valid.  Set request.user and persist user
-                        # in the session by logging the user in.
-                        auth.login(request, user)
-                        request.session[self.TOKEN_SESSION_KEY] = token
-
+                user = auth.authenticate(request, remote_user=username)
+                if user:
+                    # User is valid.  Set request.user and persist user
+                    # in the session by logging the user in.
+                    auth.login(request, user)
+                    request.session[self.TOKEN_SESSION_KEY] = token
+                    logger.info(json.dumps(
+                        {'msg': "Login token based session",
+                         'user': username}))
             except InvalidTokenError as ex:
                 return HttpResponse(status=401,
                                     reason='Invalid token: {}'.format(ex))
         else:
             if (request.user.is_authenticated and
                     self.TOKEN_SESSION_KEY in request.session):
-                # The user is authenticated with a token in session, but the
-                # request does not contain a token, the session is invalid.
-                del request.session[self.TOKEN_SESSION_KEY]
+                # The session was established based on a valid token
+                # but the request has no token, revoke the existing session.
+                logger.error("Revoke token based session: {}".format(
```

Comment:
Logging an error here will send email to the django admins if they do not add a logging handler in settings.  My preference would be to remove logging from this library.

Reply:
These logging will be useful in the first stage of the integration testing. I have a checklist item to check the logging and reduce them afterwards.